sanic/request.py

```diff
@@ -71,7 +71,7 @@ def is_full(self):
         return self._queue.full()
 
 
-class Request(dict):
+class Request:
```

Comment:
This change will break any and all current usage where someone is iterating over the request object like `for _key in request`. Is there a reason why these changes is required?

Reply:
@harshanarayana Is this actually being used? Iteration, or the entire storage feature? Request is not a dictionary, so it shouldn't inherit just to gain those features. This has led to problems as pointed out in #1309.

The iteration APIs of dict could be added, but frankly I'd rather do what @abuckenheimer suggested and make it a public property instead. If anything, I'd expect request[...] to fetch me HTTP headers (but keeping that at request.headers[...] is better, just like request.storage[...]).

Reply:
I can't tell you if it is being used or not. I mean the iterable nature of it. This was how it was working. So before we break this, it might be a good idea to give out a deprecation in the next release and then remove it.